src/bank/use-cases/find-bank-by-id.ts

```diff
@@ -0,0 +1,19 @@
+import { Injectable } from '@nestjs/common';
+
+import { Bank } from '@/bank/infra/entities';
+import { IBankRepository } from '@/bank/interfaces';
+
+@Injectable()
+export class GetBankByIdUseCase {
+  constructor(private readonly bankRepository: IBankRepository) {}
+
+  async execute(id: string): Promise<{ bank: Bank }> {
+    const bank = await this.bankRepository.findById(id);
+
+    if (!bank) {
+      throw new Error('Banco não encontrado.');
+    }
```

Comment:
Tirar as chaves e deixar em uma linha só:

    if (!bank) throw new Error('Banco não encontrado.');
    
   Esse "Error" provavelmente vai lançar um 500 se não tiver outra tratativa. 
   
   Coloca um not found (404)

Reply:
Verdade Diogo! Só nao coloquei o not found pois não haverá requisições neste repository, será apenas para consulta